lib/filterx/filterx-error.h

```diff
@@ -40,6 +40,7 @@ typedef struct _FilterXError
 
 void filterx_error_clear_error(FilterXError *error);
 EVTTAG *filterx_error_format_error(FilterXError *error);
+EVTTAG *filterx_error_format_error_location(FilterXError *error);
```

Comment:
Some of the Error related logic stayed in Eval, which is not ideal IMO, like `filterx_eval_push_error()` implements things that is the responsibility of the Error class.

We can keep it like this for now, and extract them later when needed, but the partial extraction introduced layering violations.